### PR TITLE
Setup controllers

### PR DESCRIPTION
**Pull request by rbanos-mv**
# Ruby Blog app

> The Blog app will be a classic example of a blog website. You will create a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.

### Project requirements

- **Important:** do NOT use `rails generate scaffold` command. The point of this project is to show you how to set up controllers by yourself.
- Take a look at the wireframes below. Make sure that your app handles the URLs used in the wireframes:

- For each URL you should create:

  - [x] A route.
  - [x] An action in the correct controller.
  - [x] A view file.

- [x] View files should include only placeholders written in plain HTML, e.g:

  - Here is a list of posts for a given user

- Make sure that you added code to handle **only** the 4 required URLs.
  - [x] posts#index
  - [x] posts#show
  - [x] users#index
  - [x] users#show

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+  </head>
+  <body>
+    <h1><%= @message %></h1>
+  </body>
+</html>
```
Comment (ShoiraTa) [inline app/views/posts/show.html.erb]: - [x] `<!DOCTYPE html>` is already defined in `#app/views/layouts/application.html.erb` so you don't have to redefine in each view, only  `<h1>Posts show page</h1>` would be enough.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+  </head>
+  <body>
+    <h1><%= @message %></h1>
+  </body>
+</html>
```
Comment (ShoiraTa) [inline app/views/users/index.html.erb]: - [x] Same goes here, please leave only `h1 `tag and delete the rest in all views except `app/views/layouts/application.html.erb` 😉

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class PostsController < ApplicationController
+  def index
+    @message = "Here is a list of all Posts by User with id: #{params[:user_id]}"
+  end
+
+  def show
+    @message = "Here is the specific Post with id: #{params[:id]} by User with id: #{params[:user_id]}"
+  end
```
Comment (ShoiraTa) [inline app/controllers/posts_controller.rb:8]: - [x] OPTIONAL: I can see you tried to create a message here and pass it to views. I suppose you did it for testing purposes. But keep in mind that Controllers should be skinny, they should be a bridge that connects database with views. EG: 

```
  def index
    @posts = Post.all
  end

  def show
    @post = Post.find(params[:id])
  end
```

So please avoid using controllers for other purposes, you can easily create the message text in views  😉

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
STATUS: CHANGES REQUIRED 🟡 
----
Hi @rbanos-mv ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

There are some issues that you still need to work on to go to the next project but you are almost there!

woohoo-youre-almost


 ## Required Changes  :recycle: 

_Check the comments under the review._ 

## Optional suggestions

_Check the comments under the review._ 

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/show.html.erb (hunk 1 above)
Comment (rbanos-mv) [reply app/views/posts/show.html.erb]: @ShoiraTa. Thanks for the explanation. I can't remember reading that concept in the documentation. I will make the requested change.

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (rbanos-mv) [reply app/controllers/posts_controller.rb:8]: @ShoiraTa 
Indeed. The requirements for today is that It should show a simple message. I was experimenting with passing a value from the controller to the view combined with getting the parameter value.

**Tomorrow or the day after, I suppose we will have to do what you suggests**. 

The tests I included today are not to test the actual content of the page. That will be done tomorrow. Today, I'm just testing that the controllers handle the routes for Users and Posts.

File: app/views/users/index.html.erb (hunk 2 above)
Comment (rbanos-mv) [reply app/views/users/index.html.erb]: Same as previous

**Review by ShadyShawkat — APPROVED** [review]
## Hi @rbanos-mv, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**